bootstrap/kubeadm/api/v1alpha4/kubeadmconfig_types.go

```diff
@@ -94,6 +97,35 @@ type KubeadmConfigSpec struct {
 	// For more information, refer to https://github.com/kubernetes-sigs/cluster-api/pull/2763#discussion_r397306055.
 	// +optional
 	UseExperimentalRetryJoin bool `json:"useExperimentalRetryJoin,omitempty"`
+
+	// Ignition contains Ignition specific configuration.
+	//
+	// +optional
+	Ignition *IgnitionSpec `json:"ignition,omitempty"`
+}
+
+// IgnitionSpec contains Ignition specific configuration.
+type IgnitionSpec struct {
+	// ContainerLinuxConfig contains CLC specific configuration.
+	//
+	// +optional
+	ContainerLinuxConfig *ContainerLinuxConfig `json:"containerLinuxConfig,omitempty"`
```

Comment:
`ContainerLinuxConfig` adds the ability to append additional ign config. How is this tied to containerLinux OS? Should this be a generic name or may be just embed `AdditionalConfig` and  `Strict` here?

Reply:
`ContainerLinuxConfig` is not tied to Container Linux OS (which is dead now). It is one of the Ignition "frontends". This particular one is used by Flatcar Container Linux. More here: https://github.com/kinvolk/container-linux-config-transpiler

Reply:
ok thanks! I'd propose to consider one the approaches I suggested above then if all this is doing is giving the ability to append ign config to the opinionated one.

Reply:
As I recall from earlier discussion, the idea behind the current struct naming was to distinguish customizations written in the Container Linux Config frontend language from those in another frontend language such as Butane Configs.  These are not interchangeable, and might conceivably have a different set of ancillary flags like `Strict`.

Reply:
Thanks for clarifying @bgilbert. I was not familia with Butane, How many config languages out there can be transpiled into ign config? Do you see much value on having this input as it is vs letting this to be a raw ignition input?

Reply:
I'm not aware of any other general-purpose ones.  Butane (previously called FCCT) is designed to extensibly support multiple distros and Ignition spec versions (&ge; 3.0) within the same binary and config language, and the goal is that distros will extend Butane rather than implementing their own transpiler.  So hopefully there will only be two cases: Butane moving forward, and CLCs for the Flatcar installed base.

I think there are some good arguments for either approach, but it does make sense to allow the k8s resource to be written in a user-friendly language.  Otherwise, the user would need to maintain the editable source of truth in a separate system, manually transpile it to an intermediate result, and inject that result into the resource.  So I'm +1 to the existing consensus in this PR.

Reply:
>Butane moving forward, and CLCs for the Flatcar installed base.

Once Flatcar gains Ignition v3 support, perhaps Butane will be usable for it too.

Reply:
@enxebre do you mind if I mark it as resolved?